sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shims/CreateDataSourceTableAsSelectCommandMetaShims.scala

```diff
@@ -48,33 +45,16 @@ final class CreateDataSourceTableAsSelectCommandMeta(
 
     val spark = SparkSession.active
     origProvider =
-      GpuDataSource.lookupDataSourceWithFallback(cmd.table.provider.get, spark.sessionState.conf)
-    // Note that the data source V2 always fallsback to the V1 currently.
-    // If that changes then this will start failing because we don't have a mapping.
-    gpuProvider = origProvider.getConstructor().newInstance() match {
-      case f: FileFormat if GpuOrcFileFormat.isSparkOrcFormat(f) =>
-        GpuOrcFileFormat.tagGpuSupport(this, spark, cmd.table.storage.properties, cmd.query.schema)
```

Comment:
This used to prevent replacing CreateDataSourceTableAsSelectCommand with a GPU version if we knew we couldn't replace the underlying data source.  Where is that happening now?  It seems like this will almost always replace the CreateDataSourceTableAsSelectCommand before it knows whether the data source can be replaced, and that could cause tests to "pass" thinking we replaced the write when in reality we did not replace the InsertIntoHadoopFsRelationCommand.

Reply:
> I wonder if it could be rewritten to not create the object and just do the validation itself

Everything I see it doing is checking the type of class on the object, which makes it seem a bit silly to construct an object from a class just to check what class the object is.  We already have the class, and can just check it directly.  But that's the way it worked before, and we can clean that up in a followup.